syntax = "proto3";

option csharp_namespace = "Recorder";

message TimeSynchronization {
    uint64 time = 1;
}

message WarningRecord {
    string text = 1;
}

message MouseDevice {
    string Name = 1;
    uint32 Handle = 2;
}

message KeyboardDevice {
    string name = 1;
    uint32 handle = 2;
}

message HidDevice {
    string name = 1;
    uint32 handle = 2;
}

message MouseRecord {
    int64 Time = 1;
    optional uint32 Flags = 2;
    // ButtonFlags and ButtonData are u16, could be unpacked by consumer.
    optional uint32 ButtonFlags = 3;
    optional uint32 ButtonData = 4;
    optional uint32 Buttons = 5;
    optional int32 X = 6;
    optional int32 Y = 7;
    optional uint32 Extra = 8;
}

message KeyboardRecord {
    int64 Time = 1;
    uint32 Code = 2;
    optional uint32 Flags = 3;
    optional uint32 Reserved = 4;
    uint32 VKey = 5;
    uint32 Message = 6;
    optional uint32 Extra = 7;
}

message HidRecord {
    int64 Time = 1;
    uint32 Size = 2;
    uint32 Count = 3;
    bytes Data = 4;
}